localstack/services/cloudformation/engine/template_utils.py

```diff
@@ -150,10 +150,20 @@ def resolve_condition(condition, conditions, parameters, mappings, stack_name):
                     # TODO: how are conditions references written here? as {"Condition": "ConditionA"} or via Ref?
                     # TODO: test for a boolean parameter?
                     param = parameters[v]
-                    return param[
-                        "ParameterValue"
-                    ]  # TODO: extend this logic, e.g. what about lists, other types, ... why is string interpreted as a boolean?
-                    # return parameters[v]
+                    parameter_type: str = param["ParameterType"]
+                    parameter_value = param.get("ResolvedValue") or param.get("ParameterValue")
+
+                    if parameter_type in ["CommaDelimitedList"] or parameter_type.startswith(
+                        "List<"
+                    ):
+                        return [p.strip() for p in parameter_value.split(",")]
+                    else:
+                        return parameter_value
```

Comment:
This is currently a known duplication of code. Not really necessary to refactor yet though.